docs/manual/about_dds/qos_provider.rst

```diff
@@ -0,0 +1,413 @@
+    - TopicQos      - ManualByParticipant
+
+
+.. code-block:: xml
+
+    <dds xmlns="http://www.omg.org/dds/"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="file://DDS_QoSProfile.xsd">
+        <qos_library name="myqoslib">
+            <qos_profile name="foo_profile">
+                <datareader_qos>
+                    <durability>
+                        <kind>TRANSIENT_DURABILITY_QOS</kind>
+                    </durability>
+                    <history>
+                        <kind>KEEP_LAST_HISTORY_QOS</kind>
+                        <depth>5</depth>
+                    </history>
+                </datareader_qos>
+                <datawriter_qos>
+                    <durability>
+                        <kind>TRANSIENT_DURABILITY_QOS</kind>
+                    </durability>
+                </datawriter_qos>
+                <topic_qos name="my_topic">
+                    <liveliness>
+                        <kind>AUTOMATIC_LIVELINESS_QOS</kind>
+                        <lease_duration>
+                            <sec>QOS_DURATION_INFINITY_SEC</sec>
+                        </lease_duration>
+                    </liveliness>
+                </topic_qos>
+            </qos_profile>
+            <qos_profile name="bar_profile">
+                <datawriter_qos>
+                    <durability>
+                        <kind>PERSISTENT_DURABILITY_QOS</kind>
+                    </durability>
+                </datawriter_qos>
+                <topic_qos name="my_topic">
+                    <liveliness>
+                        <kind>MANUAL_BY_PARTICIPANT_LIVALINESS</kind>
+                        <lease_duration>
+                            <sec>QOS_DURATION_INFINITY_SEC</sec>
+                        </lease_duration>
+                    </liveliness>
+                </topic_qos>
+            </qos_profile>
+        </qos_library>
+    </dds>
+
+
+Code Example
+============
+
+The following C application is an example to illustrate how the QoS settings from the above XML could be accessed.
+
+.. tabs::
+
+    .. group-tab:: C
+
+      .. code-block:: C
+
+        #include "dds/dds.h"
+        #include "dds/ddsc/dds_qos_provider.h"
+        #include "datatypes.h"
+
+        int main (int argc, char **argv)
+        {
+            (void) argc;
+            (void) argv;
+
+            // provider will contains:
+            // myqoslib::foo_profile                     READER
+            // myqoslib::foo_profile                     WRITER
+            // myqoslib::foo_profile::my_topic           TOPIC
+            // myqoslib::bar_profile                     WRITER
+            // myqoslib::bar_profile::my_topic           TOPIC
+            dds_qos_provider_t *provider;
+            dds_return_t ret = dds_create_qos_provider ("/path/to/qos_definitions.xml", &provider);
+            assert (ret == DDS_RETCODE_OK);
+
+            dds_qos_t *tp_qos, *wr_qos;
```

Comment:
But then these should be `dds_qos_t const * tp_qos` etc. If I am getting my own copy, then the lifetime issue mentioned above doesn't hold.